torch/_dynamo/optimizations/distributed.py

```diff
@@ -140,7 +140,8 @@ def compile_fn(self, gm: fx.GraphModule, example_inputs: List[torch.Tensor]):
         and returns its callable.
         """
         fake_mode = fake_mode_from_tensors(example_inputs)
-        assert fake_mode is not None
```

Comment:
Is this because no example inputs? If so should we assert no inputs?

Reply:
If you want a more stringent assert it seems better to assert all input tensors are fake; you could have non empty tensor list if they are all real. But I'm mostly trying to unblock users first